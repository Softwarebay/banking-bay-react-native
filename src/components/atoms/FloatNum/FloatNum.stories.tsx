import { object, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { CenterStory } from 'utils/storybook/CenterStory';

import { FloatNum } from './FloatNum';

const intStyleKnob = {
	fontSize: 32,
};

const floatStyleKnob = {
	fontSize: 20,
};

storiesOf('Atoms/FloatNum', module)
	.addDecorator((getStory) => <CenterStory center>{getStory()}</CenterStory>)
	.add('Default', () => (
		<FloatNum
			floatStyle={object('floatStyle', floatStyleKnob)}
			intStyle={object('intStyle', intStyleKnob)}
			number={text('number', '33.33')}
		/>
	));
