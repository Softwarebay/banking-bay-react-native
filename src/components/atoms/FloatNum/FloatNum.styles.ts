import { StyleSheet } from 'react-native';

import { colors } from 'theme';

export const styles = StyleSheet.create({
	container: {
		alignItems: 'baseline',
	},
	text: {
		color: colors.mainDark,
	},
});
