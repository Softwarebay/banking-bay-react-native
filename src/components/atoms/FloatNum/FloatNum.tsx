import { StyleProp, Text, TextStyle } from 'react-native';

import { Flex } from '../Flex/Flex';

import { styles } from './FloatNum.styles';

interface Props {
	number: string;
	prefix?: string;
	intStyle?: StyleProp<TextStyle>;
	floatStyle?: StyleProp<TextStyle>;
	style?: StyleProp<TextStyle>;
	testID?: string;
}

export const FloatNum = ({
	number,
	prefix,
	intStyle,
	floatStyle,
	style,
	testID,
}: Props) => {
	const [integer, float] = number.split('.');

	return (
		<Flex direction="row" style={[styles.container, style]} testID={testID}>
			<Text style={[styles.text, intStyle]}>{`${
				prefix ?? ''
			}${integer}.`}</Text>
			<Text style={[styles.text, floatStyle]}>{float}</Text>
		</Flex>
	);
};
