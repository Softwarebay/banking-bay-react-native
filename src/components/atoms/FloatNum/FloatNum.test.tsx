import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { FloatNum } from './FloatNum';

import type { Props } from './FloatNum';

describe('FloatNum', () => {
	const props: Omit<Props, 'prefix'> = {
		number: '20.22',
		testID: 'floatNum-test-id',
		floatStyle: { fontSize: 30 },
		intStyle: { fontSize: 50 },
		style: {
			height: 200,
		},
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<FloatNum {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render integer and float as separate text', () => {
		render(<FloatNum {...props} />);

		const floatNum = screen.getByTestId('floatNum-test-id');

		expect(floatNum.children).toHaveLength(2);
	});

	it('should apply separate styles for integer and float', () => {
		render(<FloatNum {...props} />);

		const int = screen.getByText('20.');
		const float = screen.getByText('22');

		expect(int).toHaveStyle({ fontSize: 50 });
		expect(float).toHaveStyle({ fontSize: 30 });
	});

	it('should display prefix with the same styles as integer text', () => {
		render(<FloatNum {...props} prefix="$" />);

		const int = screen.getByText(/20/);
		const prefix = screen.getByText(/\$/);

		expect(int).toHaveStyle({ fontSize: 50 });
		expect(prefix).toHaveStyle({ fontSize: 50 });
	});

	it('should apply additional styles from props', () => {
		render(<FloatNum {...props} />);

		const floatNum = screen.getByTestId('floatNum-test-id');

		expect(floatNum).toHaveStyle({ height: 200 });
	});
});
