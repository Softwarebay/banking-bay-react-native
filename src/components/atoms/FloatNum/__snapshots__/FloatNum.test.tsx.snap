// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatNum should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "row",
      },
      Array [
        Object {
          "alignItems": "baseline",
        },
        Object {
          "height": 200,
        },
      ],
    ]
  }
  testID="floatNum-test-id"
>
  <Text
    style={
      Array [
        Object {
          "color": "#1B1D4D",
        },
        Object {
          "fontSize": 50,
        },
      ]
    }
  >
    $20.
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#1B1D4D",
        },
        Object {
          "fontSize": 30,
        },
      ]
    }
  >
    22
  </Text>
</View>
`;
